memory/libhugetlbfs.py

```diff
@@ -98,11 +100,37 @@ def setUp(self):
         os.chdir(self.srcdir)
         patch = self.params.get('patch', default='elflink.patch')
         process.run('patch -p1 < %s' % data_dir + '/' + patch, shell=True)
+
+        if (detected_distro.name == "redhat" or
+                detected_distro.name == "fedora"):
+            falloc_patch = 'patch -p1 < %s ' % (
+                os.path.join(data_dir, 'falloc.patch'))
+            process.run(falloc_patch, shell=True)
+
         build.make(self.srcdir, extra_args='BUILDTYPE=NATIVEONLY')
 
+    def fail_check(self, log, check):
```

Comment:
Btw, insted of:

`parsed_results[32][key.split("*")[1].strip()] = int(values.split()[0])`

it can be like:

`parsed_results[32][key.lstrip('* ')] = int(values.split()[0])`

Easier to understand :)